### PR TITLE
Refactor Topics API

### PR DESCRIPTION
**Pull request by piyushchhabra**
Closes https://github.com/novuhq/novu-java/issues/51

### CODE REVIEW

**Comment by piyushchhabra** [comment]
Hi @mayorJAY please review this PR

File: src/main/java/co/novu/api/topics/TopicHandler.java [hunk 1]
```diff
@@ -10,63 +10,73 @@
...
-    public DeleteTopicResponse deleteTopic(String topicKey) {
-        boolean isSuccess = restHandler.handleDeleteForVoid(novuConfig, ENDPOINT + "/" + topicKey);
+    public DeleteTopicResponse deleteTopic(String topicKey) throws IOException {
+        Response<DeleteTopicResponse> response = topicApi.deleteTopic(topicKey).execute();
+        boolean isSuccess = response.isSuccessful();
         if (isSuccess) {
             return new DeleteTopicResponse();
         }
         return null;
     }
```
Comment (mayorJAY) [inline src/main/java/co/novu/api/topics/TopicHandler.java:71]: You don't need to check if it's successful here, there is a second **_extractResponse()_** function in RestHandler class. You should use that instead

File: src/main/java/co/novu/api/topics/TopicHandler.java [hunk 2]
```diff
@@ -10,63 +10,73 @@
...
+    public CheckTopicSubscriberResponse checkTopicSubscriber(String topicKey, String externalSubscriberId) throws IOException, NovuNetworkException {
+        Response<CheckTopicSubscriberResponse> response = topicApi.checkTopicSubscriber(topicKey, externalSubscriberId).execute();
+        return restHandler.extractResponse(response);
     }
 
-    public SubscriberRemovalResponse removeSubscriberFromTopic(SubscriberAdditionRequest request, String topicKey) {
-        boolean isSuccess = restHandler.handlePostForVoid(request, novuConfig, ENDPOINT + "/" + topicKey + "/subscribers/removal");
+    public SubscriberRemovalResponse removeSubscriberFromTopic(SubscriberAdditionRequest request, String topicKey) throws IOException {
+        Response<SubscriberRemovalResponse> response = topicApi.removeSubscriberFromTopic(topicKey, request).execute();
+        boolean isSuccess = response.isSuccessful();
```
Comment (mayorJAY) [inline src/main/java/co/novu/api/topics/TopicHandler.java:57]: See comment on **_deleteTopic()_**

File: src/main/java/co/novu/api/topics/TopicApi.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
+package co.novu.api.topics;
+
+import co.novu.api.topics.requests.SubscriberAdditionRequest;
+import co.novu.api.topics.requests.TopicRequest;
+import co.novu.api.topics.responses.*;
+import retrofit2.Call;
+import retrofit2.http.*;
```
Comment (mayorJAY) [inline src/main/java/co/novu/api/topics/TopicApi.java:7]: Please avoid wildcard imports

**Review by mayorJAY — APPROVED** [review]
LGTM 🚀

File: src/main/java/co/novu/api/topics/TopicApi.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    Call<SubscriberAdditionResponse> addSubscriberToTopic(@Path("topicKey") String topicKey, @Body SubscriberAdditionRequest subscriberAdditionRequest);
+
+    @GET(ENDPOINT + "/{topicKey}/subscribers/{externalSubscriberId}")
+    Call<CheckTopicSubscriberResponse> checkTopicSubscriber(@Path("topicKey") String topicKey, @Path("externalSubscriberId") String externalSubscriberId);
+
+    @POST(ENDPOINT + "/{topicKey}/subscribers/removal")
+    Call<SubscriberRemovalResponse> removeSubscriberFromTopic(@Path("topicKey") String topicKey, @Body SubscriberAdditionRequest subscriberAdditionRequest);
+
+    @DELETE(ENDPOINT + "/{topicKey}")
+    Call<DeleteTopicResponse> deleteTopic(@Path("topicKey") String topicKey);
```
Comment (mayorJAY) [inline src/main/java/co/novu/api/topics/TopicApi.java:42]: This should return Call<Void> and not Call< DeleteTopicResponse>
Please change that

**Comment by piyushchhabra** [comment]
Addressed the comments 👍 @mayorJAY 
I also did the same fix in other PR as well https://github.com/novuhq/novu-java/pull/75

**Review by mayorJAY — APPROVED** [review]
LGTM 🚀

**Pull request merged**